2.1.0 fails to build on non-Mac with Qt5

2.1.0 fails to build with Qt5 due to a change accidentally breaking source compatibility. Specifically, UIDictionary.cpp uses the Qt6-only QLibraryInfo::path after a change introduced in https://github.com/Sigil-Ebook/PageEdit/commit/6013e8be06faf1c30b7d85188b93d746196ac2e1

Error:

```
[ 75%] Building CXX object CMakeFiles/pageedit.dir/UIDictionary.cpp.o
/usr/bin/c++ -DQT_CORE_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_POSITIONING_LIB -DQT_PRINTSUPPORT_LIB -DQT_QMLMODELS_LIB -DQT_QML_LIB -DQT_QUICK_LIB -DQT_SVG_LIB -DQT_WEBCHANNEL_LIB -DQT_WEBENGINECORE_LIB -DQT_WEBENGINEWIDGETS_LIB -DQT_WEBENGINE_LIB -DQT_WIDGETS_LIB -DQT_XML_LIB -I/build/pageedit-2.1.0+dfsg/obj-x86_64-linux-gnu/pageedit_autogen/include -I/build/pageedit-2.1.0+dfsg/obj-x86_64-linux-gnu -I/build/pageedit-2.1.0+dfsg -I/build/pageedit-2.1.0+dfsg/gumbo_subtree/src -isystem /usr/include/x86_64-linux-gnu/qt5 -isystem /usr/include/x86_64-linux-gnu/qt5/QtWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtGui -isystem /usr/include/x86_64-linux-gnu/qt5/QtCore -isystem /usr/lib/x86_64-linux-gnu/qt5/mkspecs/linux-g++ -isystem /usr/include/x86_64-linux-gnu/qt5/QtXml -isystem /usr/include/x86_64-linux-gnu/qt5/QtPrintSupport -isystem /usr/include/x86_64-linux-gnu/qt5/QtWebEngine -isystem /usr/include/x86_64-linux-gnu/qt5/QtWebEngineCore -isystem /usr/include/x86_64-linux-gnu/qt5/QtQuick -isystem /usr/include/x86_64-linux-gnu/qt5/QtQmlModels -isystem /usr/include/x86_64-linux-gnu/qt5/QtQml -isystem /usr/include/x86_64-linux-gnu/qt5/QtNetwork -isystem /usr/include/x86_64-linux-gnu/qt5/QtWebChannel -isystem /usr/include/x86_64-linux-gnu/qt5/QtPositioning -isystem /usr/include/x86_64-linux-gnu/qt5/QtWebEngineWidgets -isystem /usr/include/x86_64-linux-gnu/qt5/QtSvg -g -O2 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -ffile-prefix-map=/build/pageedit-2.1.0+dfsg=. -flto=auto -ffat-lto-objects -fstack-protector-strong -fstack-clash-protection -Wformat -Werror=format-security -fcf-protection -fdebug-prefix-map=/build/pageedit-2.1.0+dfsg=/usr/src/pageedit-2.1.0+dfsg-1~ml1 -Wdate-time -D_FORTIFY_SOURCE=3   -Wall -fPIC -MD -MT CMakeFiles/pageedit.dir/UIDictionary.cpp.o -MF CMakeFiles/pageedit.dir/UIDictionary.cpp.o.d -o CMakeFiles/pageedit.dir/UIDictionary.cpp.o -c /build/pageedit-2.1.0+dfsg/UIDictionary.cpp
/build/pageedit-2.1.0+dfsg/UIDictionary.cpp: In static member function 'static QString UIDictionary::GetDictionaryPath()':
/build/pageedit-2.1.0+dfsg/UIDictionary.cpp:60:31: error: 'path' is not a member of 'QLibraryInfo'
   60 |     dict_path = QLibraryInfo::path(QLibraryInfo::DataPath) + "/qtwebengine_dictionaries";
      |                               ^~~~
make[3]: *** [CMakeFiles/pageedit.dir/build.make:556: CMakeFiles/pageedit.dir/UIDictionary.cpp.o] Error 1
```

--- a/UIDictionary.cpp
+++ b/UIDictionary.cpp
@@ -57,7 +57,11 @@ QString UIDictionary::GetDictionaryPath()
 
     // finally look inside the installed Qt directories
 #ifndef Q_OS_MAC
+#if QT_VERSION < QT_VERSION_CHECK(6, 0, 0)
+    dict_path = QLibraryInfo::location(QLibraryInfo::DataPath) + "/qtwebengine_dictionaries";
+#else
     dict_path = QLibraryInfo::path(QLibraryInfo::DataPath) + "/qtwebengine_dictionaries";
+#endif
 #else
     dict_path = QCoreApplication::applicationDirPath() + 
       "/../Frameworks/QtWebEngineCore.framework/Resources/qtwebengine_dictionaries";

